src/Tools/dotnet-monitor/Extensibility/FolderExtensionRepository.cs

```diff
@@ -0,0 +1,83 @@
+    {
+        private readonly string _targetFolder;
+        private readonly IFileProvider _fileSystem;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public FolderExtensionRepository(IFileProvider fileSystem, ILoggerFactory loggerFactory, int resolvePriority, string targetFolder)
+            : base(resolvePriority, string.Format(CultureInfo.CurrentCulture, Strings.Message_FolderExtensionRepoName, targetFolder))
+        {
+            _fileSystem = fileSystem;
+
+            // This is the root that we will start looking at relative to _fileSystem, _fileSystem is already scoped to the correct folder, so we use an empty string
+            _targetFolder = String.Empty;
+
+            _loggerFactory = loggerFactory;
+        }
+
+        public override TExtensionType FindExtension<TExtensionType>(string extensionMoniker) where TExtensionType : class
+        {
+            // Pass 1: Require an exact case sensitive match
+            IExtension exactMatch = ScanExtensionDir(extensionMoniker, StringComparison.Ordinal);
+            if (exactMatch != null)
+            {
+                TExtensionType resultExact = (TExtensionType)exactMatch;
+                return resultExact;
+            }
+
+            // Pass 2: Relax case sensitivity
+            IExtension caseVarianceMatch = ScanExtensionDir(extensionMoniker, StringComparison.OrdinalIgnoreCase);
+            if (caseVarianceMatch != null)
+            {
+                TExtensionType result = (TExtensionType)caseVarianceMatch;
+                return result;
+            }
+
+            return null;
+        }
+
+        private IExtension ScanExtensionDir(string extensionMoniker, StringComparison comparisionType)
+        {
+            IDirectoryContents extensionRoot = _fileSystem.GetDirectoryContents(_targetFolder);
+
+            string expectedFile = GetOSSpecificExecutableFormat(extensionMoniker);
+
+            foreach (IFileInfo extDir in extensionRoot)
+            {
+                if (extDir.Exists && extDir.IsDirectory && string.Equals(extDir.Name, extensionMoniker, comparisionType))
+                {
+                    IDirectoryContents extension = _fileSystem.GetDirectoryContents(_targetFolder);
+                    foreach (IFileInfo file in extension)
+                    {
+                        if (file.Exists && !file.IsDirectory && string.Equals(file.Name, expectedFile, comparisionType))
+                        {
+                            ILogger<ProgramExtension> logger = _loggerFactory.CreateLogger<ProgramExtension>();
+                            return new ProgramExtension(Path.Combine(_targetFolder, extDir.Name, file.Name), logger);
+                        }
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        private string GetOSSpecificExecutableFormat(string executableName)
+        {
+            string format = "{0}.exe";
```

Comment:
How do we deal with exe stubs for dotnet extension implementations? Cross plat builds of extensions?